src/DIRAC/FrameworkSystem/Utilities/diracx.py

```diff
@@ -26,7 +31,73 @@
 DEFAULT_TOKEN_CACHE_TTL = 5 * 60
 DEFAULT_TOKEN_CACHE_SIZE = 1024
 
-legacy_exchange_session = requests.Session()
+# Number of pools to use for a given host.
+# It should be in the order of host behind the alias
+SESSION_NUM_POOLS = 20
+# Number of connection per Pool
+SESSION_CONNECTION_POOL_MAX_SIZE = 10
+
+
+class RandomizedPoolManager(PoolManager):
+    """
+    A PoolManager subclass that creates multiple connection pools per host.
+    Each connection request randomly picks one of the available pools.
+    """
+
+    def __init__(self, num_pools=3, **kwargs):
+        self.num_pools = num_pools
+        super().__init__(**kwargs)
+
+    def connection_from_host(self, host, port=None, scheme="http", pool_kwargs=None):
+        # Pick a random index to diversify the pool key.
+
+        rand_index = random.randint(0, self.num_pools - 1)
+        pool_key = (f"{host}-{rand_index}", port, scheme)
+        if pool_key in self.pools:
+            return self.pools[pool_key]
+
+        # Create a new pool if none exists for this key.
+        if scheme == "http":
+            self.pools[pool_key] = HTTPConnectionPool(host, port, **self.connection_pool_kw)
+        elif scheme == "https":
+            self.pools[pool_key] = HTTPSConnectionPool(host, port, **self.connection_pool_kw)
+        else:
+            raise ValueError(f"Unsupported scheme: {scheme}")
+
+        return self.pools[pool_key]
+
+
+class RandomizedHTTPAdapter(HTTPAdapter):
+    """
+    An HTTPAdapter that uses the RandomizedPoolManager.
+    """
+
+    def __init__(self, num_pools=3, maxsize=10, **kwargs):
+        self.num_pools = num_pools
+        self.custom_maxsize = maxsize
+        super().__init__(**kwargs)
+
+    def init_poolmanager(self, connections, maxsize, block=False, **pool_kwargs):
+        """
+        Initialize the pool manager with our custom RandomizedPoolManager.
+        """
+        pool_kwargs.update(
+            {
+                "maxsize": self.custom_maxsize,
+                "block": block,
```

Comment:
```suggestion
                "pool_block": block,
```
? 
cf. https://requests.readthedocs.io/en/latest/api/#requests.adapters.HTTPAdapter

Also `pool_maxsize`?
Or

Reply:
No, it really is `block` and `maxsize` :man_shrugging: 
https://requests.readthedocs.io/en/latest/api/#requests.adapters.HTTPAdapter.init_poolmanager

Reply:
But you are not calling `init_poolmanager`, are you?

You are calling `__init__`?

https://github.com/DIRACGrid/DIRAC/pull/8144/files#diff-8b9db1c9898828a7dffadf105de25e0c2dfd3f38bab1d8bf69325a16c39d3fcdR49

Reply:
Ok, discussed in person.

Reply:
added the comment as discussed